plugins/modules/ndo_interface_setting.py

```diff
@@ -0,0 +1,866 @@
+
+            if min_links and match.details.get("portChannelPolicy", {}).get("minLinks") != min_links:
+                ops.append(dict(op="replace", path="{0}/{1}/portChannelPolicy/minLinks".format(path, match.index), value=min_links))
+                match.details["portChannelPolicy"]["minLinks"] = min_links
+
+            if max_links and match.details.get("portChannelPolicy", {}).get("maxLinks") != max_links:
+                ops.append(dict(op="replace", path="{0}/{1}/portChannelPolicy/maxLinks".format(path, match.index), value=max_links))
+                match.details["portChannelPolicy"]["maxLinks"] = max_links
+
+            if load_balance_hashing and match.details.get("portChannelPolicy", {}).get("hashFields") != load_balance_hashing:
+                ops.append(dict(op="replace", path="{0}/{1}/portChannelPolicy/hashFields".format(path, match.index), value=load_balance_hashing))
+                match.details["portChannelPolicy"]["hashFields"] = load_balance_hashing
+
+            if controls and match.details.get("portChannelPolicy", {}).get("control") != controls:
+                ops.append(dict(op="replace", path="{0}/{1}/portChannelPolicy/control".format(path, match.index), value=controls))
+                match.details["portChannelPolicy"]["control"] = controls
+            elif controls == []:
+                ops.append(dict(op="remove", path="{0}/{1}/portChannelPolicy/control".format(path, match.index)))
+                match.details.pop("controls", None)
+
+            mso.sanitize(match.details)
+
+        else:
+            if not interface_type:
+                mso.fail_json(msg="Error: Missing required argument 'interface_type' for creating an Interface Policy Group.")
+            payload = {
+                "name": interface_policy_group,
+                "type": interface_type,
+                "templateId": mso_template.template.get("templateId"),
+                "schemaId": mso_template.template.get("schemaId"),
+                "llfc": {},
+                "stp": {},
+                "l2Interface": {},
+                "lldp": {},
+                "linkLevel": {},
+                "mcp": {},
+                "portChannelPolicy": {},
+            }
+
+            if description:
+                payload["description"] = description
+
+            if domains:
+                domain_uuid = validate_domains(mso, domains, template, template_info)
+                payload["domains"] = domain_uuid
+
+            if synce:
+                existing_sync_e = validate_sync_e(mso, synce, template, template_info)
+                if synce in existing_sync_e:
+                    payload["syncEthPolicy"] = existing_sync_e[synce]
+
+            if access_macsec_policy:
+                existing_access_macsec_policy = validate_macsec_policy(mso, access_macsec_policy, template, template_info)
+                if access_macsec_policy in existing_access_macsec_policy:
```

Comment:
add a test case with false scenario

Reply:
As discussed, validate function checks this scenario and this additional check is not required, so removed.